spec/models/user_spec.rb

```diff
@@ -0,0 +1,28 @@
+describe User do
+
+  let(:user) do
+    create(:user)
+  end
+
+  it 'requires a matching confirmation password' do
+    user = build(:user, password_confirmation: "wrong")
+    expect { sign_up(user) }.not_to change(User, :count)
+  end
+
+  it 'authenticates when given an valid username and password' do
+   authenticated_user = User.authenticate(username: user.username, password: user.password)
+   expect(authenticated_user).to eq user
+ end
```

Comment:
`end` at 15, 1 is not aligned with `it 'authenticates when given an valid username and password' do` at 12, 2